regen/lock_definitions.pl

```diff
@@ -0,0 +1,2401 @@
+            # this decision to not list could easily be changed.
+            if (keys %races_with > 1) {
+
+                # Find myself.
+                my $myself = delete $races_with{$use};
+                my $race_text =
+                    "${output_function_name}has potential races with other"
+                  . " threads concurrently using";
+
+                # Sort the remaining ones.
+                my @race_names = sort name_order keys %races_with;
+
+                # Create a list with me as the first item.  Doing this gets
+                # the sub to create the proper text based on the real number
+                # of items; otherwise the deleted one could mislead it.
+                my $list_text = format_list(join ",", $myself, @race_names);
+
+                # Change me to the word itself.
+                $list_text =~ s/ $myself \(\)? ,? \s* //x;
+                if (@race_names == 1) {
+                    $race_text .= " either itself " . $list_text;
+                }
+                else {
+                    $race_text .= " any of: itself, " . $list_text;
+                }
+
+                $race_text = wrap($columns, "", $hanging, "$race_text.");
+                push @comments, split "\n", $race_text;
+            }
+        }
+
+        if ($entry->{conditions}) {
+            my $quoted = join(", ", $entry->{conditions}->@*);
+            $quoted = "'$quoted'" if $quoted !~ /\s/;
+            push @comments, $uses{$use}{display_name}
+                          . " locking macros are only valid if "
+                           . $quoted;
+        }
+
+        if ($entry->{notes}) {
+            foreach my $note ($entry->{notes}->@*) {
+                push @comments, split "\n", wrap($columns, "", $hanging,
+                                                $output_function_name . $note);
+            }
+        }
+
+        # Ready to output any comments
+        if (@comments) {
+            print $l "\n $dindent/* ", $comments[0];
+            for (my $i = 1; $i < @comments; $i++) {
+                print $l "\n $dindent * ", $comments[$i];
+            }
+
+            if (length($comments[-1]) + length($dindent) + 3
+                                                           <= $comment_columns)
+            {   # End the comment with a trailing "*/" if it fits on the line
+                print $l " */\n";
+            }
+            else {  # Otherwise, put it on the next line
+                print $l "\n $dindent */\n";
+            }
+        }
+
+        # Now calculate and output the wrapper macros.
+
+        if ($entry->{unsuitable}) {
+            croak("Unsuitable function '$use' has a lock")
+                                if $entry->{locks} || $entry->{categories};
+            print $l <<~EOT;
+                #${dindent}define ${USE}_LOCK                              \\
+                #${dindent}  error_${use}_not_suitable_for_multi-threaded_operation
```

Comment:
Why does this generate:
```
#  error_foo_not suitable...
```
rather than
```
#  error foo_not suitable...
```
?
The first may produce an error, but it might not include the text of the symbol.

The second will produce the requested message, though it might be better to quote it, which will prevent macro replacement, allowing you to avoid the need for `all_the_underscores`.